app/perftools/management/commands/create_page_cache.py

```diff
@@ -49,6 +49,285 @@
 
 logger = logging.getLogger(__name__)
 
+
+def fetch_jtbd_hackathons():
+    db = JSONStore.objects.get(key='hackathons', view='hackathons')
+    status = db.data[0]
+    hackathons = db.data[1][0:2]
+    fields = ['logo', 'name', 'slug', 'summary', 'start_date', 'end_date', 'sponsor_profiles']
+    return [{k: v for k, v in event.items() if k in fields} for event in hackathons if status == 'upcoming']
+
+
+def create_jtbd_earn_cache():
+    print('create_jtbd_earn_cache')
+    import datetime
+    from bleach import clean
+    from app.utils import ellipses
+    from marketing.models import LeaderboardRank
+
+    leaderboard = LeaderboardRank.objects.filter(
+        active=True, product='bounties', leaderboard='monthly_earners',
+    ).order_by('-amount')[0:4].cache()
+
+    thirty_days_ago = timezone.now() - datetime.timedelta(days=30)
+
+    bounties_qs = Bounty.objects.current().filter(
+        network='mainnet', event=None, idx_status='open', created_on__gt=thirty_days_ago
+    ).order_by('-_val_usd_db')[0:2]
+
+    top_earners = bounties = []
+
+    for earner in leaderboard:
+        top_earners.append({
+            'rank': earner.rank,
+            'amount': earner.amount,
+            'avatar_url': earner.avatar_url,
+            'username': earner.github_username,
+        })
```

Comment:
instead of looping maybe is better to do the query using values or values_list:

`LeaderboardRank.objects.filter(
        active=True, product='bounties', leaderboard='monthly_earners',
    ).values('rank', 'amount', 'avatar_url', 'username').order_by('-amount')[0:4].cache()`
    
check the docs https://docs.djangoproject.com/en/3.1/ref/models/querysets/#values
> Returns a QuerySet that returns dictionaries, rather than model instances, when used as an iterable.

  https://docs.djangoproject.com/en/3.1/ref/models/querysets/#values-list

Reply:
this will throw and error cause `avatar_url` is a property not a field. @octavioamu

Reply:
for reference

@octavioamu said

```
well you can always go with github_username and call the avatar from the html

check marketing/utils.py
def get_platform_wide_stats(since_last_n_days=90):
L291
```